Hw3 ftp

AntonYermilov opened the pull request.

AntonYermilov committed:
hw3-ftp: init commit
added Client and Server implementation

AntonYermilov committed:
hw3-ftp:
Updated build.gradle
Added javadocs to Client
Minor fixes in Server
Added files for testing

AntonYermilov committed:
hw3-ftp
Added javadocs to Server
Minor fixes in Client

AntonYermilov committed:
hw3-ftp
Updated idea config files

AntonYermilov committed:
Updated javadocs.
Improved Client and Server implementation, added logging
Added tests

jzuken commented on term2/hw3-ftp/src/test/java/me/eranik/ftp/InteractionTest.java:
```diff
@@ -0,0 +1,185 @@
+
+        String result = "9\n" +
+                "src/test/resources/1 true\n" +
+                "src/test/resources/1/2 true\n" +
```
раз уж вы пишете на кроссплатформенном языке, давайте сделаем так, чтобы тесты тоже проходили под разными ОС 
вот тут, например, описано несколько вариантов этого достичь: https://www.sghill.net/how-do-i-make-cross-platform-file-paths-in-java.html

jzuken commented on term2/hw3-ftp/src/main/java/me/eranik/ftp/Server.java:
```diff
@@ -0,0 +1,224 @@
+     * @param args list of arguments: first argument contains port number
+     */
+    public static void main(String[] args) {
+        int portNumber = Integer.parseInt(args[0]);
```
а если аргументы забыли?

jzuken commented on term2/hw3-ftp/src/main/java/me/eranik/ftp/Client.java:
```diff
@@ -0,0 +1,159 @@
+                System.exit(0);
+            }
+
+            if (!type.equals("1") && !type.equals("2") || !reader.hasNext()) {
```
привязать пользовательский интерфейс к протоколу -- так себе идея. я думаю, пользователи заслужили чуть более человеческих команд

jzuken commented on term2/hw3-ftp/src/main/java/me/eranik/ftp/Client.java:
```diff
@@ -0,0 +1,159 @@
+                if (type == 1) {
+                    listFiles(dataInput);
+                } else {
+                    saveFile(dataInput, new File(path).getName());
```
хорошо бы как-то уведомлять клиента об успешности завершения операций

AntonYermilov committed:
hw3-ftp:
Added argument checker
Added messages about errors and successful query processing.
Tests are cross-platform now

jzuken commented on term2/hw3-ftp/src/main/java/me/eranik/ftp/Server.java:
```diff
@@ -0,0 +1,229 @@
+            System.exit(1);
+        }
+
+        int portNumber = Integer.parseInt(args[0]);
```
ну а если введут что-то не так, оно падает с NumberFormatException. неужто так сложно сразу подумать, что может пойти не так?
-1

jzuken commented on term2/hw3-ftp/src/main/java/me/eranik/ftp/Client.java:
```diff
@@ -0,0 +1,186 @@
+                    break;
+                case SUCCESS:
+                    if (type.equals("get")) {
+                        writer.println("File was successfully downloaded to the current directory.");
```
get .\src\test
File was successfully downloaded to the current directory.

да вот как-то не очень. вы либо запрещайте качать каталоги, либо качайте их как-то адекватно
-1

jzuken commented on term2/hw3-ftp/src/main/java/me/eranik/ftp/Server.java:
```diff
@@ -0,0 +1,229 @@
+                    output.flush();
+                    return;
+                }
+                if (type == 2) {
```
магические константы. вообще лучше детали протокола переиспользовать между клиентом и сервером

AntonYermilov committed:
hw3-ftp
Fixed fails when reading incorrect arguments
Fixed messages about successful downloads when file was not found
Added tests
Added interaction protocol enums
